src/core/MOM_dynamics_split_RK2.F90

```diff
@@ -860,6 +888,93 @@ subroutine step_MOM_dyn_split_RK2(u, v, h, tv, visc, Time_local, dt, forces, p_s
+  enddo ; enddo
+
+  if (CS%id_hfPFu > 0) then
+    do k=1,nz ; do j=js,je ; do I=Isq,Ieq
+      hf_PFu(I,j,k) = CS%PFu(I,j,k) * CS%diag_hfrac_u(I,j,k)
+    enddo ; enddo ; enddo
+    call post_data(CS%id_hfPFu, hf_PFu, CS%diag)
+  endif
+  if (CS%id_hfPFv > 0) then
+    do k=1,nz ; do J=Jsq,Jeq ; do i=is,ie
+      hf_PFv(i,J,k) = CS%PFv(i,J,k) * CS%diag_hfrac_v(i,J,k)
+    enddo ; enddo ; enddo
+    call post_data(CS%id_hfPFv, hf_PFv, CS%diag)
+  endif
+  if (CS%id_hfPFu_2d > 0) then
+    do k=1,nz ; do j=js,je ; do I=Isq,Ieq
+      hf_PFu_2d(I,j) = hf_PFu_2d(I,j) + CS%PFu(I,j,k) * CS%diag_hfrac_u(I,j,k)
+    enddo ; enddo ; enddo
+    call post_data(CS%id_hfPFu_2d, hf_PFu_2d, CS%diag)
+  endif
+  if (CS%id_hfPFv_2d > 0) then
+    do k=1,nz ; do J=Jsq,Jeq ; do i=is,ie
+      hf_PFv_2d(i,J) = hf_PFv_2d(i,J) + CS%PFv(i,J,k) * CS%diag_hfrac_v(i,J,k)
+    enddo ; enddo ; enddo
+    call post_data(CS%id_hfPFv_2d, hf_PFv_2d, CS%diag)
+  endif
+
+  if (CS%id_hfCAu > 0) then
+    do k=1,nz ; do j=js,je ; do I=Isq,Ieq
+      hf_CAu(I,j,k) = CS%CAu(I,j,k) * CS%diag_hfrac_u(I,j,k)
+    enddo ; enddo ; enddo
+    call post_data(CS%id_hfCAu, hf_CAu, CS%diag)
+  endif
+  if (CS%id_hfCAv > 0) then
+    do k=1,nz ; do J=Jsq,Jeq ; do i=is,ie
+      hf_CAv(i,J,k) = CS%CAv(i,J,k) * CS%diag_hfrac_v(i,J,k)
+    enddo ; enddo ; enddo
+    call post_data(CS%id_hfCAv, hf_CAv, CS%diag)
+  endif
+  if (CS%id_hfCAu_2d > 0) then
+    do k=1,nz ; do j=js,je ; do I=Isq,Ieq
+      hf_CAu_2d(I,j) = hf_CAu_2d(I,j) + CS%CAu(I,j,k) * CS%diag_hfrac_u(I,j,k)
+    enddo ; enddo ; enddo
+    call post_data(CS%id_hfCAu_2d, hf_CAu_2d, CS%diag)
+  endif
+  if (CS%id_hfCAv_2d > 0) then
+    do k=1,nz ; do J=Jsq,Jeq ; do i=is,ie
+      hf_CAv_2d(i,J) = hf_CAv_2d(i,J) + CS%CAv(i,J,k) * CS%diag_hfrac_v(i,J,k)
+    enddo ; enddo ; enddo
+    call post_data(CS%id_hfCAv_2d, hf_CAv_2d, CS%diag)
+  endif
+
+  if (CS%id_hfu_BT_accel > 0) then
+    do k=1,nz ; do j=js,je ; do I=Isq,Ieq
+      hf_uBT_accel(I,j,k) = CS%u_accel_bt(I,j,k) * CS%diag_hfrac_u(I,j,k)
+    enddo ; enddo ; enddo
+    call post_data(CS%id_hfu_BT_accel, hf_uBT_accel, CS%diag)
+  endif
+  if (CS%id_hfv_BT_accel > 0) then
+    do k=1,nz ; do J=Jsq,Jeq ; do i=is,ie
+      hf_vBT_accel(i,J,k) = CS%v_accel_bt(i,J,k) * CS%diag_hfrac_v(i,J,k)
+    enddo ; enddo ; enddo
+    call post_data(CS%id_hfv_BT_accel, hf_vBT_accel, CS%diag)
+  endif
+  if (CS%id_hfu_BT_accel_2d > 0) then
+    do k=1,nz ; do j=js,je ; do I=Isq,Ieq
+      hf_uBT_accel_2d(I,j) = hf_uBT_accel_2d(I,j) + CS%u_accel_bt(I,j,k) * CS%diag_hfrac_u(I,j,k)
+    enddo ; enddo ; enddo
+    call post_data(CS%id_hfu_BT_accel_2d, hf_uBT_accel_2d, CS%diag)
+  endif
+  if (CS%id_hfv_BT_accel_2d > 0) then
+    do k=1,nz ; do J=Jsq,Jeq ; do i=is,ie
+      hf_vBT_accel_2d(i,J) = hf_vBT_accel_2d(i,J) + CS%v_accel_bt(i,J,k) * CS%diag_hfrac_v(i,J,k)
+    enddo ; enddo ; enddo
+    call post_data(CS%id_hfv_BT_accel_2d, hf_vBT_accel_2d, CS%diag)
+  endif
+
```

Comment:
Array variable names and id variable names should be the same: e.g. `id_hfv_BT_accel` and `hf_vBT_accel`, should be renamed to: `id_hf_vBT_accel`